src/compile/axis/rules.ts

```diff
@@ -123,3 +124,10 @@ export function zindex(specifiedAxis: Axis, isGridAxis: boolean) {
   }
   return 1; // otherwise return undefined and use Vega's default.
 };
+
+export function facet(property: keyof VgAxis, specifiedAxis: Axis, isGridAxis: boolean, channel: Channel) {
```

Comment:
`facet` is a pretty confusing name.   This is a common rules for `domain` and `ticks`, which deal with their rules no matter you have facet or not.   

If you want a single method, a better name would be `domainAndTicks`.
For clean API, I would not export `domainAndTicks` and instead 
```
export const domain = domainAndTicks;
export const ticks = domainAndTicks;
```

and just call `domain` and `ticks` from `axis/parse`. 

This way if we wanna customize domain / ticks in the future, we only need to change them inside this file.  

(`axis/parse` shouldn't need to know that rules for these two properties are the same.)